### DISCUSSION TITLE
Display the bot version number in PR description footer

### DISCUSSION THREAD

**Discussion by astorije** [renovatebot/renovate#24832] [Suggest an Idea]
### What would you like Renovate to be able to do?

This is the current footer of a PR opened by Renovate:

<img width="495" alt="Screen Shot 2023-02-22 at 1 03 34 PM" src="https://user-images.githubusercontent.com/113730/220716403-a34900a6-17ce-41e7-bbdf-034f3b38e279.png">

It would be great to see which version of the Renovate bot opened a given PR:

<img width="519" alt="Screen Shot 2023-02-22 at 1 05 10 PM" src="https://user-images.githubusercontent.com/113730/220716877-cd70ccd4-caa1-47e9-9560-ad867278e02a.png">

_Additional context: Our (self-hosted) bot is maintained by our DevOps team through parts of infrastructure that repository owners have no access to. It's not easy (or possible depending on privileges) to check the logs, and we are now using the `latest` Docker image (making it even harder as we can't just check a piece of configuration on a repo somewhere), so having this directly in the PR would be really neat and helpful. Even with full access to logs, it's nice to have this out there._

### If you have any ideas on how this should be implemented, please tell us here.

As far as I understand, the PR footer can be configured with: https://github.com/renovatebot/renovate/blob/main/lib/config/options/index.ts#L1802
Is there a way to template this string with, for example: `This PR has been generated by [Renovate Bot](https://github.com/renovatebot/renovate) version {{renovateBotVersion}}`?

I don't believe so if I get the code right, I'm wondering if one way to do it could be to edit https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/update/pr/body/footer.ts#L6 to:

```ts
return '\n---\n\n' + template.safeCompile(config.prFooter, { ...config, renovateBotVersion: '???' });
```

Of course, I realize this is probably too hacky, so let me know if there is a better way to set this up. I don't believe the bot version should be added to the `BranchConfig` type because it's not a value users should configure themselves.

Is there an environment variable or an object/function I could import somewhere to retrieve the version the bot is currently running?

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by rarkins** [comment]
If you look at the source of that footer, you see something like

```
This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://app.renovatebot.com/dashboard#github/renovatebot/renovate).
<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNC4xMDguNCIsInVwZGF0ZWRJblZlciI6IjM0LjEwOC40In0=-->
```

If you base64 decode that debug string you see `{"createdInVer":"34.108.4","updatedInVer":"34.108.4"}`

**Comment by HonkingGoose** [comment]
We already discussed something similar a few weeks back, but we didn't create an issue for it. 😄 

In short I proposed we show to the Renovate users in the Dependency Dashboard: [^issue]

- The runtime/program version of Renovate that ran
- The "behavior" version that's used

We should show that information in the PR's as well.

[^issue]: https://github.com/renovatebot/renovate/issues/19980

**Comment by astorije** [comment]
> If you base64 decode that debug string you see `{"createdInVer":"34.108.4","updatedInVer":"34.108.4"}`

Aaah this is cool, I had no idea, thank you! I still think it would be great to see the version number in PR footers, but that's a nice trick/backup!

> The "behavior" version that's used

What's that?

> We should show that information in the PR's as well.

Clever to put this in the dependency dashboard, didn't think of that! Though I do think there is value to having this in PR descriptions as well, since over time PRs will be opened with different versions.

**Comment by HonkingGoose** [comment]
> > The "behavior" version that's used
> 
> What's that?

I should have explained this better! 🙈 The one-line summary is:

> Provide a way to make backwards-incompatible ("breaking") changes to Renovation functionality but maintain old behavior for existing orgs and/or repos.

Read the top post of this issue to learn more:

- #19980

**Comment by HonkingGoose** [comment]
Summarizing the discussion. It looks like we want to show the:

- Renovate version that **created** the PR
- Renovate version that last **updated** the PR
- _Optional_ Last run version of "behavior version" (in case user opted-in to "old behavior) #19980

@rarkins can you review my summary and tell us what you want to do?

**Comment by rarkins** [comment]
I'm not really sure. I think that the information is already there, but it's encoded. So one possibility would be to have it raw/unencoded so users can inspects it more easily.
